ls1intum/Artemis
# Use generic delete dialog for resetting, cleanup

### Checklist

### Motivation and Context


This should resolve the issue #874. Archiving will be replaced by Downloading Repos functionality both on Course Administration and View pages for Programming Exercise.

### Description

We have a lot of code duplication in creating the dialog, as well as separate popup URLs for each dialog. This PBI is focused on resolving this issues. Additionally, the dialog should wait until the delete request is executed and not close if the error is thrown.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Select Reset button on Programming Exercise (1).
4. Select Cleanup button on Programming Exercise (2).
5. Select Reset button on Quiz Exercise (3).
6. Select Delete button on Programming Exercise.
7. Click 'Delete' in the dialog, it should take a bit of time until the request to server is sent and we get the result in the client (4).

### Screenshots


#### Reset Programming Exercise (Course Administration)
![Screenshot_2019-11-23 Courses](https://user-images.githubusercontent.com/17262376/69485920-d0647c00-0e45-11ea-9458-a62a83c2d8ab.png)
---
#### Cleanup Programming Exercise (Course Administration and Programming View pages)
![Screenshot_2019-10-14 Courses(1)](https://user-images.githubusercontent.com/17262376/66778147-29e29e00-eecb-11e9-91f6-acafc789b84f.png)
---
#### Reset Quiz Exercise (Course Administration)
![Screenshot_2019-10-14 Courses(2)](https://user-images.githubusercontent.com/17262376/66778151-2bac6180-eecb-11e9-8a8e-69ba08feed0c.png)
---
#### Download Repos button on Programming Exercise(Course Administration)
